{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Hyper params"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!pip install git+https://github.com/huggingface/transformers\n",
    "\n",
    "# The model that you want to train from the Hugging Face hub\n",
    "model_name = \"codellama/CodeLlama-7b-Instruct-hf\"\n",
    "\n",
    "# The instruction dataset to use\n",
    "dataset_name = \"pechaut/cairo-instruct\"\n",
    "\n",
    "#\n",
    "# Fine-tuned model name\n",
    "new_model = \"llama-2-7b-cairo-test\"\n",
    "\n",
    "# Output directory where the model predictions and checkpoints will be stored\n",
    "output_dir = \"./results\"\n",
    "\n",
    "import os\n",
    "os.environ[\"TOKENIZERS_PARALLELISM\"] = \"false\"\n",
    "\n",
    "hub_name = \"pechaut/codellama-cairo-instruct\"\n",
    "\n",
    "max_steps = 1000"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Load Base Model and Tokenizer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "881c0e8037594342a70d97a80877f13b",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Loading checkpoint shards:   0%|          | 0/2 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "1"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import torch\n",
    "from transformers import (\n",
    "    AutoTokenizer,\n",
    "    AutoModelForCausalLM,\n",
    "    BitsAndBytesConfig,\n",
    ")\n",
    "\n",
    "\n",
    "# load the quantized settings: 4 bit quantization\n",
    "bnb_config = BitsAndBytesConfig(\n",
    "    load_in_4bit=True,\n",
    "    bnb_4bit_quant_type=\"nf4\",\n",
    "    bnb_4bit_compute_dtype=torch.bfloat16,\n",
    "    bnb_4bit_use_double_quant=True,\n",
    ")\n",
    "\n",
    "# Load base model\n",
    "model = AutoModelForCausalLM.from_pretrained(\n",
    "    model_name,\n",
    "    quantization_config=bnb_config,\n",
    "    device_map={\"\": 0}\n",
    ")\n",
    "\n",
    "# don't use the cache\n",
    "model.config.use_cache = False\n",
    "\n",
    "# Load the tokenizer from the model (llama2)\n",
    "tokenizer = AutoTokenizer.from_pretrained(model_name)\n",
    "tokenizer.add_special_tokens({'pad_token': '[PAD]'})"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Import Dataset from Hugging Face"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/pec/miniconda3/envs/cairo-code/lib/python3.9/site-packages/datasets/load.py:2097: FutureWarning: 'ignore_verifications' was deprecated in favor of 'verification_mode' in version 2.9.1 and will be removed in 3.0.0.\n",
      "You can remove this warning by passing 'verification_mode=no_checks' instead.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "6144e307531d44d9ad06fcab37234e76",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Downloading readme:   0%|          | 0.00/379 [00:00<?, ?B/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "3d74f10ad1b5491788300ee84ca03a4d",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Downloading data files:   0%|          | 0/2 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "90a81ba403fd4037bc286194de4dec33",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Downloading data:   0%|          | 0.00/2.42k [00:00<?, ?B/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "786e12b1ac754d65a1b82eff9cd06ed9",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Downloading data:   0%|          | 0.00/29.1k [00:00<?, ?B/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "9c53b2d9b59f4be79dcf99d067dba11a",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Extracting data files:   0%|          | 0/2 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "c04cbd3752364607b497b5858e9a453f",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Generating eval split:   0%|          | 0/5 [00:00<?, ? examples/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "949b35c90a7c4c849e5adea52863cb3d",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Generating train split:   0%|          | 0/197 [00:00<?, ? examples/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "0db7810c7e2b48a690f5d0072690c619",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Downloading readme:   0%|          | 0.00/379 [00:00<?, ?B/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "790bee668443444c9b0ab94b6a8e1abf",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Downloading data files:   0%|          | 0/2 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "78c2c6845b634a35a62cfefb9cfd5ef3",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Downloading data:   0%|          | 0.00/2.42k [00:00<?, ?B/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "2a6b96dec7b542ec80404e693e312ae9",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Downloading data:   0%|          | 0.00/29.1k [00:00<?, ?B/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "4d25e0af28ee4743b49a9f69d3963130",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Extracting data files:   0%|          | 0/2 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "4bfc3e32e7884a678fe8396f5505ac59",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Generating eval split:   0%|          | 0/5 [00:00<?, ? examples/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "1fd0754971b442a983fcffa8aea21011",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Generating train split:   0%|          | 0/197 [00:00<?, ? examples/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from datasets import load_dataset\n",
    "\n",
    "# Load the dataset\n",
    "dataset_train = load_dataset(dataset_name, split=\"train\", download_mode='force_redownload',ignore_verifications=True)\n",
    "dataset_test = load_dataset(dataset_name, split=\"eval\", download_mode='force_redownload',ignore_verifications=True)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Check to see if everything is fine before launching training"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<s>[INST]In Cairo:  test if myArray is empty[/INST]\n",
      "    if(myArry.is_empty) 'it is empty'.print()\n",
      "\n",
      " Explanations:\n",
      "\n",
      " - If you want to check if an array is empty or not, you can use the is_empty() method, which returns true if the array is empty and false otherwise.\n",
      " </s>\n"
     ]
    }
   ],
   "source": [
    "df = dataset_train.to_pandas()\n",
    "print(df[\"text\"][18])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Load Model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/pec/miniconda3/envs/cairo-code/lib/python3.9/site-packages/peft/utils/other.py:133: FutureWarning: prepare_model_for_int8_training is deprecated and will be removed in a future version. Use prepare_model_for_kbit_training instead.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "488bc503c3234865b09cb1e47da88a3b",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Map:   0%|          | 0/197 [00:00<?, ? examples/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "4fdf4afbb55d4da69413c0e36a6bdd15",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Map:   0%|          | 0/5 [00:00<?, ? examples/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/pec/miniconda3/envs/cairo-code/lib/python3.9/site-packages/trl/trainer/sft_trainer.py:207: UserWarning: You passed a tokenizer with `padding_side` not equal to `right` to the SFTTrainer. This might lead to some unexpected behaviour due to overflow issues when training a model in half-precision. You might consider adding `tokenizer.padding_side = 'right'` to your code.\n",
      "  warnings.warn(\n"
     ]
    }
   ],
   "source": [
    "from trl import SFTTrainer\n",
    "import transformers\n",
    "from peft import LoraConfig\n",
    "\n",
    "qlora_config = LoraConfig(\n",
    "    r=16,\n",
    "    lora_alpha=32,\n",
    "    lora_dropout=0.05,\n",
    "    bias=\"none\",\n",
    "    task_type=\"CAUSAL_LM\"\n",
    ")\n",
    "\n",
    "supervised_finetuning_trainer = SFTTrainer(\n",
    "    model,\n",
    "    train_dataset=dataset_train,\n",
    "    eval_dataset=dataset_test,\n",
    "    args=transformers.TrainingArguments(\n",
    "        per_device_train_batch_size=1,\n",
    "        gradient_accumulation_steps=4,\n",
    "        learning_rate=2e-4,\n",
    "        max_steps=max_steps,\n",
    "        logging_steps=10,\n",
    "        output_dir=\"./SFTOpenLM-Dolly15k\",\n",
    "        optim=\"paged_adamw_8bit\",\n",
    "        fp16=True,\n",
    "    ),\n",
    "    tokenizer=tokenizer,\n",
    "    peft_config=qlora_config,\n",
    "    dataset_text_field=\"text\",\n",
    "    max_seq_length=512\n",
    ")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "You're using a CodeLlamaTokenizerFast tokenizer. Please note that with a fast tokenizer, using the `__call__` method is faster than using a method to encode the text followed by a call to the `pad` method to get a padded encoding.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "\n",
       "    <div>\n",
       "      \n",
       "      <progress value='113' max='1000' style='width:300px; height:20px; vertical-align: middle;'></progress>\n",
       "      [ 113/1000 16:56 < 2:15:19, 0.11 it/s, Epoch 2.27/21]\n",
       "    </div>\n",
       "    <table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       " <tr style=\"text-align: left;\">\n",
       "      <th>Step</th>\n",
       "      <th>Training Loss</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>3.140900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>2.067300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>1.752600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>40</td>\n",
       "      <td>1.602200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>50</td>\n",
       "      <td>1.432800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>60</td>\n",
       "      <td>1.379600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>70</td>\n",
       "      <td>1.359900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>80</td>\n",
       "      <td>1.098900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>90</td>\n",
       "      <td>1.152700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>100</td>\n",
       "      <td>1.110200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>110</td>\n",
       "      <td>0.889300</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table><p>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "supervised_finetuning_trainer.train()\n",
    "supervised_finetuning_trainer.model.save_pretrained(new_model)\n",
    "supervised_finetuning_trainer.tokenizer.save_pretrained(new_model)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Load the old model, apply the new model and push to hub"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "2bc90107c0e3498d894884618bf11fe8",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Loading checkpoint shards:   0%|          | 0/2 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.\n",
      "Using pad_token, but it is not set yet.\n",
      "Using pad_token, but it is not set yet.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CommitInfo(commit_url='https://huggingface.co/pechaut/codellama-cairo-instruct/commit/c7e337195258c59af4cdd70738639480b06e3a83', commit_message='Upload tokenizer', commit_description='', oid='c7e337195258c59af4cdd70738639480b06e3a83', pr_url=None, pr_revision=None, pr_num=None)"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from peft import AutoPeftModelForCausalLM, PeftModel\n",
    "from transformers import AutoModelForCausalLM\n",
    "import torch\n",
    "import os\n",
    "model = AutoModelForCausalLM.from_pretrained(\n",
    "            model_name, torch_dtype=torch.float16, load_in_8bit=False, device_map=\"auto\", trust_remote_code=True)\n",
    "     \n",
    "peft_model = PeftModel.from_pretrained(model, new_model, from_transformers=True)\n",
    "     \n",
    "\n",
    "model = peft_model.merge_and_unload()\n",
    "\n",
    "model.push_to_hub(hub_name)\n",
    "from transformers import AutoTokenizer\n",
    "tokenizer = AutoTokenizer.from_pretrained(new_model)\n",
    "tokenizer.push_to_hub(hub_name)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Define a pipe ready to use"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.\n"
     ]
    },
    {
     "ename": "RuntimeError",
     "evalue": "Error(s) in loading state_dict for LlamaForCausalLM:\n\tWhile copying the parameter named \"model.layers.0.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.0.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.0.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.0.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.0.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.0.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.0.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.0.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.0.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.1.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.1.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.1.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.1.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.1.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.1.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.1.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.1.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.1.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.2.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.2.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.2.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.2.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.2.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.2.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.2.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.2.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.2.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.3.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.3.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.3.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.3.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.3.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.3.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.3.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.3.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.3.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.4.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.4.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.4.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.4.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.4.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.4.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.4.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.4.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.4.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.5.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.5.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.5.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.5.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.5.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.5.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.5.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.5.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.5.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.6.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.6.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.6.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.6.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.6.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.6.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.6.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.6.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.6.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.7.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.7.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.7.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.7.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.7.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.7.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.7.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.7.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.7.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.8.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.8.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.8.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.8.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.8.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.8.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.8.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.8.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.8.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.9.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.9.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.9.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.9.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.9.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.9.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.9.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.9.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.9.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.10.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.10.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.10.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.10.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.10.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.10.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.10.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.10.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.10.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.11.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.11.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.11.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.11.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.11.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.11.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.11.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.11.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.11.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.12.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.12.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.12.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.12.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.12.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.12.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.12.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.12.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.12.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.13.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.13.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.13.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.13.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.13.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.13.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.13.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.13.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.13.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.14.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.14.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.14.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.14.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.14.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.14.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.14.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.14.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.14.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.15.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.15.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.15.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.15.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.15.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.15.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.15.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.15.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.15.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.16.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.16.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.16.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.16.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.16.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.16.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.16.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.16.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.16.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.17.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.17.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.17.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.17.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.17.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.17.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.17.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.17.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.17.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.18.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.18.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.18.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.18.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.18.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.18.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.18.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.18.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.18.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.19.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.19.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.19.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.19.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.19.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.19.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.19.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.19.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.19.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.20.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.20.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.20.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.20.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.20.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.20.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.20.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.20.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.20.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.21.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.21.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.21.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.21.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.21.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.21.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.21.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.21.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.21.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.22.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.22.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.22.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.22.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.22.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.22.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.22.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.22.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.22.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.23.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.23.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.23.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.23.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.23.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.23.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.23.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.23.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.23.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.24.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.24.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.24.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.24.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.24.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.24.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.24.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.24.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.24.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.25.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.25.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.25.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.25.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.25.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.25.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.25.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.25.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.25.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.26.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.26.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.26.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.26.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.26.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.26.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.26.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.26.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.26.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.27.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.27.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.27.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.27.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.27.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.27.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.27.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.27.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.27.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.28.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.28.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.28.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.28.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.28.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.28.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.28.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.28.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.28.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.29.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.29.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.29.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.29.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.29.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.29.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.29.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.29.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.29.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.30.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.30.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.30.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.30.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.30.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.30.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.30.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.30.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.30.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.31.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.31.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.31.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.31.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.31.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.31.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.31.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.31.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.31.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.norm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"lm_head.weight\", whose dimensions in the model are torch.Size([32016, 4096]) and whose dimensions in the checkpoint are torch.Size([32016, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mRuntimeError\u001b[0m                              Traceback (most recent call last)",
      "\u001b[1;32m/home/pec/cairo-code/cairo-code - llama.ipynb Cellule 13\u001b[0m line \u001b[0;36m6\n\u001b[1;32m      <a href='vscode-notebook-cell://wsl%2Bubuntu/home/pec/cairo-code/cairo-code%20-%20llama.ipynb#X14sdnNjb2RlLXJlbW90ZQ%3D%3D?line=2'>3</a>\u001b[0m \u001b[39mimport\u001b[39;00m \u001b[39mtorch\u001b[39;00m\n\u001b[1;32m      <a href='vscode-notebook-cell://wsl%2Bubuntu/home/pec/cairo-code/cairo-code%20-%20llama.ipynb#X14sdnNjb2RlLXJlbW90ZQ%3D%3D?line=4'>5</a>\u001b[0m tokenizer \u001b[39m=\u001b[39m AutoTokenizer\u001b[39m.\u001b[39mfrom_pretrained(hub_name)\n\u001b[0;32m----> <a href='vscode-notebook-cell://wsl%2Bubuntu/home/pec/cairo-code/cairo-code%20-%20llama.ipynb#X14sdnNjb2RlLXJlbW90ZQ%3D%3D?line=5'>6</a>\u001b[0m pipe \u001b[39m=\u001b[39m pipeline(\n\u001b[1;32m      <a href='vscode-notebook-cell://wsl%2Bubuntu/home/pec/cairo-code/cairo-code%20-%20llama.ipynb#X14sdnNjb2RlLXJlbW90ZQ%3D%3D?line=6'>7</a>\u001b[0m     \u001b[39m\"\u001b[39;49m\u001b[39mtext-generation\u001b[39;49m\u001b[39m\"\u001b[39;49m,\n\u001b[1;32m      <a href='vscode-notebook-cell://wsl%2Bubuntu/home/pec/cairo-code/cairo-code%20-%20llama.ipynb#X14sdnNjb2RlLXJlbW90ZQ%3D%3D?line=7'>8</a>\u001b[0m     model\u001b[39m=\u001b[39;49mhub_name,\n\u001b[1;32m      <a href='vscode-notebook-cell://wsl%2Bubuntu/home/pec/cairo-code/cairo-code%20-%20llama.ipynb#X14sdnNjb2RlLXJlbW90ZQ%3D%3D?line=8'>9</a>\u001b[0m     torch_dtype\u001b[39m=\u001b[39;49mtorch\u001b[39m.\u001b[39;49mfloat16,\n\u001b[1;32m     <a href='vscode-notebook-cell://wsl%2Bubuntu/home/pec/cairo-code/cairo-code%20-%20llama.ipynb#X14sdnNjb2RlLXJlbW90ZQ%3D%3D?line=9'>10</a>\u001b[0m     \n\u001b[1;32m     <a href='vscode-notebook-cell://wsl%2Bubuntu/home/pec/cairo-code/cairo-code%20-%20llama.ipynb#X14sdnNjb2RlLXJlbW90ZQ%3D%3D?line=10'>11</a>\u001b[0m )\n",
      "File \u001b[0;32m~/miniconda3/envs/cairo-code/lib/python3.9/site-packages/transformers/pipelines/__init__.py:834\u001b[0m, in \u001b[0;36mpipeline\u001b[0;34m(task, model, config, tokenizer, feature_extractor, image_processor, framework, revision, use_fast, token, device, device_map, torch_dtype, trust_remote_code, model_kwargs, pipeline_class, **kwargs)\u001b[0m\n\u001b[1;32m    832\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39misinstance\u001b[39m(model, \u001b[39mstr\u001b[39m) \u001b[39mor\u001b[39;00m framework \u001b[39mis\u001b[39;00m \u001b[39mNone\u001b[39;00m:\n\u001b[1;32m    833\u001b[0m     model_classes \u001b[39m=\u001b[39m {\u001b[39m\"\u001b[39m\u001b[39mtf\u001b[39m\u001b[39m\"\u001b[39m: targeted_task[\u001b[39m\"\u001b[39m\u001b[39mtf\u001b[39m\u001b[39m\"\u001b[39m], \u001b[39m\"\u001b[39m\u001b[39mpt\u001b[39m\u001b[39m\"\u001b[39m: targeted_task[\u001b[39m\"\u001b[39m\u001b[39mpt\u001b[39m\u001b[39m\"\u001b[39m]}\n\u001b[0;32m--> 834\u001b[0m     framework, model \u001b[39m=\u001b[39m infer_framework_load_model(\n\u001b[1;32m    835\u001b[0m         model,\n\u001b[1;32m    836\u001b[0m         model_classes\u001b[39m=\u001b[39;49mmodel_classes,\n\u001b[1;32m    837\u001b[0m         config\u001b[39m=\u001b[39;49mconfig,\n\u001b[1;32m    838\u001b[0m         framework\u001b[39m=\u001b[39;49mframework,\n\u001b[1;32m    839\u001b[0m         task\u001b[39m=\u001b[39;49mtask,\n\u001b[1;32m    840\u001b[0m         \u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49mhub_kwargs,\n\u001b[1;32m    841\u001b[0m         \u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49mmodel_kwargs,\n\u001b[1;32m    842\u001b[0m     )\n\u001b[1;32m    844\u001b[0m model_config \u001b[39m=\u001b[39m model\u001b[39m.\u001b[39mconfig\n\u001b[1;32m    845\u001b[0m hub_kwargs[\u001b[39m\"\u001b[39m\u001b[39m_commit_hash\u001b[39m\u001b[39m\"\u001b[39m] \u001b[39m=\u001b[39m model\u001b[39m.\u001b[39mconfig\u001b[39m.\u001b[39m_commit_hash\n",
      "File \u001b[0;32m~/miniconda3/envs/cairo-code/lib/python3.9/site-packages/transformers/pipelines/base.py:269\u001b[0m, in \u001b[0;36minfer_framework_load_model\u001b[0;34m(model, config, model_classes, task, framework, **model_kwargs)\u001b[0m\n\u001b[1;32m    263\u001b[0m     logger\u001b[39m.\u001b[39mwarning(\n\u001b[1;32m    264\u001b[0m         \u001b[39m\"\u001b[39m\u001b[39mModel might be a PyTorch model (ending with `.bin`) but PyTorch is not available. \u001b[39m\u001b[39m\"\u001b[39m\n\u001b[1;32m    265\u001b[0m         \u001b[39m\"\u001b[39m\u001b[39mTrying to load the model with Tensorflow.\u001b[39m\u001b[39m\"\u001b[39m\n\u001b[1;32m    266\u001b[0m     )\n\u001b[1;32m    268\u001b[0m \u001b[39mtry\u001b[39;00m:\n\u001b[0;32m--> 269\u001b[0m     model \u001b[39m=\u001b[39m model_class\u001b[39m.\u001b[39;49mfrom_pretrained(model, \u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49mkwargs)\n\u001b[1;32m    270\u001b[0m     \u001b[39mif\u001b[39;00m \u001b[39mhasattr\u001b[39m(model, \u001b[39m\"\u001b[39m\u001b[39meval\u001b[39m\u001b[39m\"\u001b[39m):\n\u001b[1;32m    271\u001b[0m         model \u001b[39m=\u001b[39m model\u001b[39m.\u001b[39meval()\n",
      "File \u001b[0;32m~/miniconda3/envs/cairo-code/lib/python3.9/site-packages/transformers/models/auto/auto_factory.py:565\u001b[0m, in \u001b[0;36m_BaseAutoModelClass.from_pretrained\u001b[0;34m(cls, pretrained_model_name_or_path, *model_args, **kwargs)\u001b[0m\n\u001b[1;32m    563\u001b[0m \u001b[39melif\u001b[39;00m \u001b[39mtype\u001b[39m(config) \u001b[39min\u001b[39;00m \u001b[39mcls\u001b[39m\u001b[39m.\u001b[39m_model_mapping\u001b[39m.\u001b[39mkeys():\n\u001b[1;32m    564\u001b[0m     model_class \u001b[39m=\u001b[39m _get_model_class(config, \u001b[39mcls\u001b[39m\u001b[39m.\u001b[39m_model_mapping)\n\u001b[0;32m--> 565\u001b[0m     \u001b[39mreturn\u001b[39;00m model_class\u001b[39m.\u001b[39;49mfrom_pretrained(\n\u001b[1;32m    566\u001b[0m         pretrained_model_name_or_path, \u001b[39m*\u001b[39;49mmodel_args, config\u001b[39m=\u001b[39;49mconfig, \u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49mhub_kwargs, \u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49mkwargs\n\u001b[1;32m    567\u001b[0m     )\n\u001b[1;32m    568\u001b[0m \u001b[39mraise\u001b[39;00m \u001b[39mValueError\u001b[39;00m(\n\u001b[1;32m    569\u001b[0m     \u001b[39mf\u001b[39m\u001b[39m\"\u001b[39m\u001b[39mUnrecognized configuration class \u001b[39m\u001b[39m{\u001b[39;00mconfig\u001b[39m.\u001b[39m\u001b[39m__class__\u001b[39m\u001b[39m}\u001b[39;00m\u001b[39m for this kind of AutoModel: \u001b[39m\u001b[39m{\u001b[39;00m\u001b[39mcls\u001b[39m\u001b[39m.\u001b[39m\u001b[39m__name__\u001b[39m\u001b[39m}\u001b[39;00m\u001b[39m.\u001b[39m\u001b[39m\\n\u001b[39;00m\u001b[39m\"\u001b[39m\n\u001b[1;32m    570\u001b[0m     \u001b[39mf\u001b[39m\u001b[39m\"\u001b[39m\u001b[39mModel type should be one of \u001b[39m\u001b[39m{\u001b[39;00m\u001b[39m'\u001b[39m\u001b[39m, \u001b[39m\u001b[39m'\u001b[39m\u001b[39m.\u001b[39mjoin(c\u001b[39m.\u001b[39m\u001b[39m__name__\u001b[39m\u001b[39m \u001b[39m\u001b[39mfor\u001b[39;00m\u001b[39m \u001b[39mc\u001b[39m \u001b[39m\u001b[39min\u001b[39;00m\u001b[39m \u001b[39m\u001b[39mcls\u001b[39m\u001b[39m.\u001b[39m_model_mapping\u001b[39m.\u001b[39mkeys())\u001b[39m}\u001b[39;00m\u001b[39m.\u001b[39m\u001b[39m\"\u001b[39m\n\u001b[1;32m    571\u001b[0m )\n",
      "File \u001b[0;32m~/miniconda3/envs/cairo-code/lib/python3.9/site-packages/transformers/modeling_utils.py:3307\u001b[0m, in \u001b[0;36mPreTrainedModel.from_pretrained\u001b[0;34m(cls, pretrained_model_name_or_path, config, cache_dir, ignore_mismatched_sizes, force_download, local_files_only, token, revision, use_safetensors, *model_args, **kwargs)\u001b[0m\n\u001b[1;32m   3297\u001b[0m     \u001b[39mif\u001b[39;00m dtype_orig \u001b[39mis\u001b[39;00m \u001b[39mnot\u001b[39;00m \u001b[39mNone\u001b[39;00m:\n\u001b[1;32m   3298\u001b[0m         torch\u001b[39m.\u001b[39mset_default_dtype(dtype_orig)\n\u001b[1;32m   3300\u001b[0m     (\n\u001b[1;32m   3301\u001b[0m         model,\n\u001b[1;32m   3302\u001b[0m         missing_keys,\n\u001b[1;32m   3303\u001b[0m         unexpected_keys,\n\u001b[1;32m   3304\u001b[0m         mismatched_keys,\n\u001b[1;32m   3305\u001b[0m         offload_index,\n\u001b[1;32m   3306\u001b[0m         error_msgs,\n\u001b[0;32m-> 3307\u001b[0m     ) \u001b[39m=\u001b[39m \u001b[39mcls\u001b[39;49m\u001b[39m.\u001b[39;49m_load_pretrained_model(\n\u001b[1;32m   3308\u001b[0m         model,\n\u001b[1;32m   3309\u001b[0m         state_dict,\n\u001b[1;32m   3310\u001b[0m         loaded_state_dict_keys,  \u001b[39m# XXX: rename?\u001b[39;49;00m\n\u001b[1;32m   3311\u001b[0m         resolved_archive_file,\n\u001b[1;32m   3312\u001b[0m         pretrained_model_name_or_path,\n\u001b[1;32m   3313\u001b[0m         ignore_mismatched_sizes\u001b[39m=\u001b[39;49mignore_mismatched_sizes,\n\u001b[1;32m   3314\u001b[0m         sharded_metadata\u001b[39m=\u001b[39;49msharded_metadata,\n\u001b[1;32m   3315\u001b[0m         _fast_init\u001b[39m=\u001b[39;49m_fast_init,\n\u001b[1;32m   3316\u001b[0m         low_cpu_mem_usage\u001b[39m=\u001b[39;49mlow_cpu_mem_usage,\n\u001b[1;32m   3317\u001b[0m         device_map\u001b[39m=\u001b[39;49mdevice_map,\n\u001b[1;32m   3318\u001b[0m         offload_folder\u001b[39m=\u001b[39;49moffload_folder,\n\u001b[1;32m   3319\u001b[0m         offload_state_dict\u001b[39m=\u001b[39;49moffload_state_dict,\n\u001b[1;32m   3320\u001b[0m         dtype\u001b[39m=\u001b[39;49mtorch_dtype,\n\u001b[1;32m   3321\u001b[0m         is_quantized\u001b[39m=\u001b[39;49m(\u001b[39mgetattr\u001b[39;49m(model, \u001b[39m\"\u001b[39;49m\u001b[39mquantization_method\u001b[39;49m\u001b[39m\"\u001b[39;49m, \u001b[39mNone\u001b[39;49;00m) \u001b[39m==\u001b[39;49m QuantizationMethod\u001b[39m.\u001b[39;49mBITS_AND_BYTES),\n\u001b[1;32m   3322\u001b[0m         keep_in_fp32_modules\u001b[39m=\u001b[39;49mkeep_in_fp32_modules,\n\u001b[1;32m   3323\u001b[0m     )\n\u001b[1;32m   3325\u001b[0m model\u001b[39m.\u001b[39mis_loaded_in_4bit \u001b[39m=\u001b[39m load_in_4bit\n\u001b[1;32m   3326\u001b[0m model\u001b[39m.\u001b[39mis_loaded_in_8bit \u001b[39m=\u001b[39m load_in_8bit\n",
      "File \u001b[0;32m~/miniconda3/envs/cairo-code/lib/python3.9/site-packages/transformers/modeling_utils.py:3756\u001b[0m, in \u001b[0;36mPreTrainedModel._load_pretrained_model\u001b[0;34m(cls, model, state_dict, loaded_keys, resolved_archive_file, pretrained_model_name_or_path, ignore_mismatched_sizes, sharded_metadata, _fast_init, low_cpu_mem_usage, device_map, offload_folder, offload_state_dict, dtype, is_quantized, keep_in_fp32_modules)\u001b[0m\n\u001b[1;32m   3752\u001b[0m     \u001b[39mif\u001b[39;00m \u001b[39m\"\u001b[39m\u001b[39msize mismatch\u001b[39m\u001b[39m\"\u001b[39m \u001b[39min\u001b[39;00m error_msg:\n\u001b[1;32m   3753\u001b[0m         error_msg \u001b[39m+\u001b[39m\u001b[39m=\u001b[39m (\n\u001b[1;32m   3754\u001b[0m             \u001b[39m\"\u001b[39m\u001b[39m\\n\u001b[39;00m\u001b[39m\\t\u001b[39;00m\u001b[39mYou may consider adding `ignore_mismatched_sizes=True` in the model `from_pretrained` method.\u001b[39m\u001b[39m\"\u001b[39m\n\u001b[1;32m   3755\u001b[0m         )\n\u001b[0;32m-> 3756\u001b[0m     \u001b[39mraise\u001b[39;00m \u001b[39mRuntimeError\u001b[39;00m(\u001b[39mf\u001b[39m\u001b[39m\"\u001b[39m\u001b[39mError(s) in loading state_dict for \u001b[39m\u001b[39m{\u001b[39;00mmodel\u001b[39m.\u001b[39m\u001b[39m__class__\u001b[39m\u001b[39m.\u001b[39m\u001b[39m__name__\u001b[39m\u001b[39m}\u001b[39;00m\u001b[39m:\u001b[39m\u001b[39m\\n\u001b[39;00m\u001b[39m\\t\u001b[39;00m\u001b[39m{\u001b[39;00merror_msg\u001b[39m}\u001b[39;00m\u001b[39m\"\u001b[39m)\n\u001b[1;32m   3758\u001b[0m \u001b[39mif\u001b[39;00m is_quantized:\n\u001b[1;32m   3759\u001b[0m     unexpected_keys \u001b[39m=\u001b[39m [elem \u001b[39mfor\u001b[39;00m elem \u001b[39min\u001b[39;00m unexpected_keys \u001b[39mif\u001b[39;00m \u001b[39m\"\u001b[39m\u001b[39mSCB\u001b[39m\u001b[39m\"\u001b[39m \u001b[39mnot\u001b[39;00m \u001b[39min\u001b[39;00m elem]\n",
      "\u001b[0;31mRuntimeError\u001b[0m: Error(s) in loading state_dict for LlamaForCausalLM:\n\tWhile copying the parameter named \"model.layers.0.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.0.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.0.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.0.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.0.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.0.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.0.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.0.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.0.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.1.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.1.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.1.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.1.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.1.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.1.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.1.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.1.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.1.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.2.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.2.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.2.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.2.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.2.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.2.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.2.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.2.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.2.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.3.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.3.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.3.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.3.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.3.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.3.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.3.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.3.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.3.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.4.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.4.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.4.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.4.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.4.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.4.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.4.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.4.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.4.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.5.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.5.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.5.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.5.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.5.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.5.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.5.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.5.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.5.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.6.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.6.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.6.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.6.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.6.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.6.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.6.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.6.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.6.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.7.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.7.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.7.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.7.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.7.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.7.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.7.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.7.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.7.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.8.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.8.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.8.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.8.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.8.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.8.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.8.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.8.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.8.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.9.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.9.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.9.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.9.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.9.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.9.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.9.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.9.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.9.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.10.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.10.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.10.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.10.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.10.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.10.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.10.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.10.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.10.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.11.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.11.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.11.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.11.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.11.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.11.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.11.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.11.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.11.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.12.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.12.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.12.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.12.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.12.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.12.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.12.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.12.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.12.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.13.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.13.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.13.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.13.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.13.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.13.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.13.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.13.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.13.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.14.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.14.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.14.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.14.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.14.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.14.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.14.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.14.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.14.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.15.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.15.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.15.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.15.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.15.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.15.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.15.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.15.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.15.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.16.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.16.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.16.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.16.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.16.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.16.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.16.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.16.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.16.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.17.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.17.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.17.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.17.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.17.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.17.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.17.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.17.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.17.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.18.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.18.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.18.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.18.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.18.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.18.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.18.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.18.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.18.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.19.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.19.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.19.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.19.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.19.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.19.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.19.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.19.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.19.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.20.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.20.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.20.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.20.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.20.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.20.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.20.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.20.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.20.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.21.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.21.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.21.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.21.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.21.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.21.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.21.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.21.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.21.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.22.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.22.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.22.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.22.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.22.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.22.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.22.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.22.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.22.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.23.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.23.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.23.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.23.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.23.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.23.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.23.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.23.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.23.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.24.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.24.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.24.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.24.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.24.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.24.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.24.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.24.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.24.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.25.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.25.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.25.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.25.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.25.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.25.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.25.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.25.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.25.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.26.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.26.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.26.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.26.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.26.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.26.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.26.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.26.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.26.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.27.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.27.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.27.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.27.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.27.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.27.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.27.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.27.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.27.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.28.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.28.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.28.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.28.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.28.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.28.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.28.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.28.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.28.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.29.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.29.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.29.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.29.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.29.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.29.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.29.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.29.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.29.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.30.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.30.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.30.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.30.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.30.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.30.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.30.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.30.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.30.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.31.self_attn.q_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.31.self_attn.k_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.31.self_attn.v_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.31.self_attn.o_proj.weight\", whose dimensions in the model are torch.Size([4096, 4096]) and whose dimensions in the checkpoint are torch.Size([4096, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.31.mlp.gate_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.31.mlp.up_proj.weight\", whose dimensions in the model are torch.Size([11008, 4096]) and whose dimensions in the checkpoint are torch.Size([11008, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.31.mlp.down_proj.weight\", whose dimensions in the model are torch.Size([4096, 11008]) and whose dimensions in the checkpoint are torch.Size([4096, 11008]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.31.input_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.layers.31.post_attention_layernorm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"model.norm.weight\", whose dimensions in the model are torch.Size([4096]) and whose dimensions in the checkpoint are torch.Size([4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',).\n\tWhile copying the parameter named \"lm_head.weight\", whose dimensions in the model are torch.Size([32016, 4096]) and whose dimensions in the checkpoint are torch.Size([32016, 4096]), an exception occurred : ('Cannot copy out of meta tensor; no data!',)."
     ]
    },
    {
     "ename": "",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31mLe Kernel s’est bloqué lors de l’exécution du code dans la cellule active ou une cellule précédente. Veuillez vérifier le code dans la ou les cellules pour identifier une cause possible de l’échec. Cliquez <a href='https://aka.ms/vscodeJupyterKernelCrash'>ici</a> pour plus d’informations. Pour plus d’informations, consultez Jupyter <a href='command:jupyter.viewOutput'>log</a>."
     ]
    }
   ],
   "source": [
    "from transformers import AutoTokenizer\n",
    "from transformers import pipeline\n",
    "import torch\n",
    "\n",
    "tokenizer = AutoTokenizer.from_pretrained(hub_name)\n",
    "pipe = pipeline(\n",
    "    \"text-generation\",\n",
    "    model=hub_name,\n",
    "    torch_dtype=torch.float16,\n",
    "    \n",
    ")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Setting `pad_token_id` to `eos_token_id`:2 for open-end generation.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Result: fn fibonacci(n: int) -> int:\n",
      "    if n <= 1:\n",
      "        return n\n",
      "    else:\n",
      "        return fibonacci(n-1) + fibonacci(n-2)\n",
      "\n",
      "\n",
      "def fibonacci_memo(n: int) -> int:\n",
      "    if n <= 1:\n",
      "        return n\n",
      "    else:\n",
      "        return fibonacci_memo(n-1) + fibonacci_memo(n-2)\n",
      "\n",
      "\n",
      "def fibonacci_memo_recursive(n: int, memo: dict = None) -> int:\n",
      "    if memo is None:\n",
      "        memo = {}\n",
      "    if n <= 1:\n",
      "        return n\n",
      "    else:\n",
      "        if n not in memo:\n",
      "            memo[n] = fibonacci_memo_recursive(n-1, memo)\n"
     ]
    }
   ],
   "source": [
    "prompt = \"create a structure mailAccount, print it\"\n",
    "\n",
    "\n",
    "\n",
    "sequences = pipe(\n",
    "   f\"[INST]I'm working in Cairo. {prompt} [/INST]\",\n",
    "    do_sample=True,\n",
    "    temperature=0.01,\n",
    "    top_k=50,\n",
    "    top_p=0.9,\n",
    "    num_return_sequences=1,\n",
    "    eos_token_id=tokenizer.eos_token_id,\n",
    "    max_length=200,\n",
    ")\n",
    "\n",
    "for seq in sequences:\n",
    "    print(f\"Result: {seq['generated_text']}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Setting `pad_token_id` to `eos_token_id`:2 for open-end generation.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[INST]I'm working in Cairo. create a structure mailAccount, print it [/INST]\n",
      "    \n",
      "struct MailAccount {\n",
      "    name: u32,\n",
      "    password: u32,\n",
      "}\n",
      "\n",
      "fn main() {\n",
      "    let mut myAccount = MailAccount {\n",
      "        name: 1,\n",
      "        password: 2,\n",
      "    };\n",
      "    // do some stuff with myAccount\n",
      "    std::mem::drop(myAccount);\n",
      "    myAccount = MailAccount {\n",
      "        name: 3,\n",
      "        password: 4,\n",
      "    };\n",
      "    // do some stuff with myAccount again\n",
      "}\n",
      "\n",
      "Explanation:\n",
      "- This program declares a struct MailAccount\n",
      "- The first time myAccount is used, it is mut\n",
      "- The second time myAccount is used, it is not mut\n",
      "\n",
      "    \n"
     ]
    }
   ],
   "source": [
    "# Ignore warnings\n",
    "\n",
    "\n",
    "#prompt = \"\"\"Create an array array of some animal names and then print the size of it\"\"\"\n",
    "#prompt=\"create an array containing some animal names, and print the length of the array, in cairo\"\n",
    "#prompt=\"in Cairo, define a u32\"\n",
    "#prompt=\"in cairo, write a 'hello world' program\"\n",
    "prompt = \"create a structure mailAccount, print it\"\n",
    "#prompt = \"create an array A containing 3  animal names. Create an array B, add all the elements of A to B, assert if the length of B is incorrect, create a span of B\"\n",
    "#prompt = \"in cairo, create a program that computes the fibonacci sequence,give samples of the result \"\n",
    "pipe = pipeline(task=\"text-generation\", model=model, tokenizer=tokenizer, max_length=200)\n",
    "result = pipe(f\"[INST]I'm working in Cairo. {prompt} [/INST]\")\n",
    "print(result[0]['generated_text'])"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "trch",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.18"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
